Purchases/Purchasing/TransactionsFactory.swift

```diff
@@ -14,17 +14,41 @@
 
 import Foundation
 
-class TransactionsFactory {
-
-    func nonSubscriptionTransactions(withSubscriptionsData subscriptionsData: [String: [[String: Any]]],
-                                     dateFormatter: DateFormatterType) -> [Transaction] {
-        subscriptionsData.flatMap { (productId: String, transactionData: [[String: Any]]) -> [Transaction] in
-            transactionData.map {
-                Transaction(with: $0, productId: productId, dateFormatter: dateFormatter)
-            }.compactMap { $0 }
-        }.sorted {
-            $0.purchaseDate < $1.purchaseDate
-        }
+enum TransactionsFactory {
+
+    static func nonSubscriptionTransactions(withSubscriptionsData subscriptionsData: [String: [[String: Any]]],
+                                            dateFormatter: DateFormatterType) -> [StoreTransaction] {
+        subscriptionsData
+            .flatMap { (productId: String, transactionData: [[String: Any]]) -> [StoreTransaction] in
+                transactionData
+                    .map { SimpleTransaction(with: $0, productID: productId, dateFormatter: dateFormatter) }
+                    .compactMap { $0 }
+                    .map { StoreTransaction($0) }
```

Comment:
[Just to know] I imagine that we should not worry about the performance of this code because a user does not have too many subscriptions. Also, I see that the transformation is done in three steps and makes it more identifiable what is done in each one, but we could simplify them, couldn't we? At least, the first `map` could be a `compactMap`, so we could eliminate the second step and reduce iterations?

Reply:
That was like that before so I didn't think about it, but good call! Changed!

Reply:
I also made it `lazy` to avoid unnecessary copies of the Arrays.